src/TextMateSharp/Internal/Parser/Json/JSONPListParser.cs

```diff
@@ -18,46 +17,85 @@ public T Parse(StreamReader contents)
         {
             PList<T> pList = new PList<T>(theme);
 
-            JsonReader reader = new JsonTextReader(contents);
+            var buffer = new byte[2048];
+
+            JsonReaderOptions options = new JsonReaderOptions()
+            {
+                AllowTrailingCommas = true,
+                CommentHandling = JsonCommentHandling.Skip
+            };
+
+            int size = contents.BaseStream.Read(buffer, 0, buffer.Length);
+            var reader = new Utf8JsonReader(buffer.AsSpan(0, size), isFinalBlock: false, state: new JsonReaderState(options));
 
             while (true)
             {
-                if (!reader.Read())
+                int bytesRead = -1;
+
+                while (!reader.Read())
+                {
+                    bytesRead = GetMoreBytesFromStream(contents.BaseStream, ref buffer, ref reader);
+
+                    if (bytesRead == 0)
+                        break;
+                }
+
+                if (bytesRead == 0)
                     break;
 
-                JsonToken nextToken = reader.TokenType;
+                var nextToken = reader.TokenType;
                 switch (nextToken)
                 {
-                    case JsonToken.StartArray:
+                    case JsonTokenType.StartArray:
                         pList.StartElement("array");
                         break;
-                    case JsonToken.EndArray:
+                    case JsonTokenType.EndArray:
                         pList.EndElement("array");
                         break;
-                    case JsonToken.StartObject:
+                    case JsonTokenType.StartObject:
                         pList.StartElement("dict");
                         break;
-                    case JsonToken.EndObject:
+                    case JsonTokenType.EndObject:
                         pList.EndElement("dict");
                         break;
-                    case JsonToken.PropertyName:
+                    case JsonTokenType.PropertyName:
                         pList.StartElement("key");
-                        pList.AddString((string)reader.Value);
+                        pList.AddString(reader.GetString());
                         pList.EndElement("key");
                         break;
-                    case JsonToken.String:
+                    case JsonTokenType.String:
                         pList.StartElement("string");
-                        pList.AddString((string)reader.Value);
+                        pList.AddString(reader.GetString());
```

Comment:
It might be possible to use CopyString instead of GetString here as well (pList.AddString appends the data to a StringBuilder, and it might be possible to use the array version of Append instead of the string version to reduce the number of temporary strings it creates)